cmd/kubectl-gadget/main.go

```diff
@@ -104,38 +103,49 @@ func main() {
 		os.Exit(1)
 	}
 
-	if !isHelp && !isDeployUndeploy && !runtimeGlobalParams.Get(grpcruntime.ParamGadgetNamespace).IsSet() {
+	if !runtimeGlobalParams.Get(grpcruntime.ParamGadgetNamespace).IsSet() {
 		gadgetNamespaces, err := utils.GetRunningGadgetNamespaces()
 		if err != nil {
-			log.Fatalf("Searching for running Inspektor Gadget instances: %s", err)
-		}
-
-		switch len(gadgetNamespaces) {
-		case 0:
-			if !isVersion {
-				log.Fatalf("No running Inspektor Gadget instances found")
+			if isHelp {
+				log.Warnf("Failed to contact Inspektor Gadget. Help text could be incomplete.")
+			} else {
+				log.Fatalf("Searching for running Inspektor Gadget instances.%s", err)
+			}
+		} else {
+			switch len(gadgetNamespaces) {
+			case 0:
+				if isHelp {
+					log.Warn("No running Inspektor Gadget instances found. Help text could be incomplete.")
+				} else {
```

Comment:
```suggestion
				} else if !isVersion {
```

We need a separate switch for `isVersion` or we will not print out the client version if we didn't deploy IG yet.
Before:
```
./kubectl-gadget version
Client version: v0.27.0-202-g8d405757
Server version: not installed
```
After:
```
./kubectl-gadget version
FATA[0000] No running Inspektor Gadget instances found. 
```

Reply:
ohman, covering all corner cases in this PR is difficult. I adjusted the logic again. Let's hope this time it's right.